examples/ensemble/plot_gradient_boosting_quantile.py

```diff
@@ -3,77 +3,268 @@
 xx = np.atleast_2d(np.linspace(0, 10, 1000)).T
 xx = xx.astype(np.float32)
 
-alpha = 0.95
+# %%
+# Plot the true function (expected mean), the prediction of the conditional
+# mean (least squares loss), the conditional median and the conditional 90%
+# interval (from 5th to 95th conditional percentiles).
+import matplotlib.pyplot as plt
 
-clf = GradientBoostingRegressor(loss='quantile', alpha=alpha,
-                                n_estimators=250, max_depth=3,
-                                learning_rate=.1, min_samples_leaf=9,
-                                min_samples_split=9)
 
-clf.fit(X, y)
+y_pred = all_models['ls'].predict(xx)
+y_lower = all_models['q 0.05'].predict(xx)
+y_upper = all_models['q 0.95'].predict(xx)
+y_med = all_models['q 0.50'].predict(xx)
 
-# Make the prediction on the meshed x-axis
-y_upper = clf.predict(xx)
+fig = plt.figure(figsize=(10, 10))
+plt.plot(xx, f(xx), 'g:', label=r'$f(x) = x\,\sin(x)$')
+plt.plot(X_test, y_test, 'b.', markersize=10, label='Test observations')
+plt.plot(xx, y_med, 'r-', label='Predicted median', color="orange")
+plt.plot(xx, y_pred, 'r-', label='Predicted mean')
+plt.plot(xx, y_upper, 'k-')
+plt.plot(xx, y_lower, 'k-')
+plt.fill_between(xx.ravel(), y_lower, y_upper, alpha=0.5,
+                 label='Predicted 90% interval')
+plt.xlabel('$x$')
+plt.ylabel('$f(x)$')
+plt.ylim(-10, 30)
+plt.legend(loc='upper left')
+plt.show()
 
-clf.set_params(alpha=1.0 - alpha)
-clf.fit(X, y)
+# %%
+# Note that the predicted median is on average below the predicted mean as the
+# noise is skewed towards high values (large outliers). Also note that the
+# median estimate is smoother because of its natural robustness to outliers.
+#
+# Also observe that the inductive bias of gradient boosting trees is
+# unfortunately preventing our 0.05 quantile to fully capture the sinoisoidal
+# shape of the signal, in particular around x=8. Tuning hyper-parameters can
+# reduce this effect as shown in the last part of this notebook.
+#
+# Analysis of the error metrics
+# -----------------------------
+#
+# Measure the models with :func:`mean_square_error` and :func:`pinball_loss`
+# metrics on the training dataset.
+from pandas import DataFrame
 
-# Make the prediction on the meshed x-axis
-y_lower = clf.predict(xx)
+results = []
+for name, gbr in sorted(all_models.items()):
+    metrics = {'model': name}
+    y_pred = gbr.predict(X_train)
+    for alpha in [0.05, 0.5, 0.95]:
+        metrics["pbl=%1.2f" % alpha] = pinball_loss(
+            y_train, y_pred, alpha=alpha)
+    metrics['MSE'] = mean_squared_error(y_train, y_pred)
+    results.append(metrics)
+DataFrame(results).set_index('model')
 
-clf.set_params(loss='ls')
-clf.fit(X, y)
+# %%
+# One column shows all models evaluated by the same metric. The minimum number
+# on a column should be obtained when the model is trained and measured with
+# the same metric. This should be always the case on the training set if the
+# training converged.
```

Comment:
Can we add that this is confirmed in out table? Would it be possible to highlight (boldface or color) the minimum value per column?

Reply:
I found a way to highlight the minimum of each column in bold.